geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/QueryInterceptor.scala

```diff
@@ -40,6 +46,14 @@ trait QueryInterceptor extends Closeable {
     * @param query query
     */
   def rewrite(query: Query): Unit
+
+  /**
+   * Hook to allow interception of a query after extracting the query values
+   *
+   * @param strategy query strategy
+   * @return an exception if the query should be stopped
+   */
+  def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = None
```

Comment:
Separately, I wonder if the QueryGuard should be its own interface?

Reply:
seems easier to combine them since we already have all the configuration wired up and they are very related.